Source/MonoGame.Extended/Timers/ContinuousClock.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.Xna.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace MonoGame.Extended.Timers
+{
+    public class ContinuousClock : IGameComponent
+    {
+        public event EventHandler ReportTimePassed;
+
+        public TimerState TimerState;
+        public double ReportEveryXSeconds;
+
+        public TimeSpan CurrentTime { get; protected set; } = TimeSpan.Zero;
+        public TimeSpan NextReport { get; protected set; } = TimeSpan.Zero;
```

Comment:
I love C# 6.0 but I'm not sure how many people are using it yet. I'd rather keep the project compatible with VS2013 for a while until there's more adoption. If you're using [Resharper](https://www.jetbrains.com/resharper/) you can set the preferred language version on a project level. I realize not everyone's going to be using it though.